src/metabase/models/field_values.clj

```diff
@@ -106,36 +106,50 @@
             (not (isa? (keyword base-type) :type/Temporal))
             (#{:list :auto-list} (keyword has-field-values)))))))
 
-(defn- values-less-than-total-max-length?
-  "`true` if the combined length of all the values in [[distinct-values]] is below the threshold for what we'll allow in a
-  FieldValues entry. Does some logging as well."
-  [distinct-values]
-  ;; only consume enough values to determine whether the total length is > `total-max-length` -- if it is, we can stop
-  (let [total-length (reduce
-                      (fn [total-length v]
-                        (let [new-total (+ total-length (count (str v)))]
-                          (if (>= new-total total-max-length)
-                            (reduced new-total)
-                            new-total)))
-                      0
-                      distinct-values)]
-    (u/prog1 (<= total-length total-max-length)
-      (log/debug (trs "Field values total length is > {0}." total-max-length)
-                 (if <>
-                   (trs "FieldValues are allowed for this Field.")
-                   (trs "FieldValues are NOT allowed for this Field."))))))
+
+(defn take-by-length
+  "Returns a transducer that takes a collection and returns a collection where the total length of all items in list is less than `max-length`."
```

Comment:
It's not, it's just fun to write 😆.
I added a 2-arity with lazy-seq version and use it instead. 
but imma keep the transducer version in case we want to use it later.